test/Mono.Linker.Tests/TestCasesRunner/TestCaseMetadaProvider.cs

```diff
@@ -164,16 +164,23 @@ public virtual IEnumerable<string> GetCommonReferencedAssemblies (NPath workingD
 		{
 			yield return workingDirectory.Combine ("Mono.Linker.Tests.Cases.Expectations.dll").ToString ();
 #if NETCOREAPP
-			string frameworkDir = Path.GetDirectoryName (typeof (object).Assembly.Location);
-
-			yield return typeof (object).Assembly.Location;
-			yield return Path.Combine (frameworkDir, "System.Runtime.dll");
-			yield return Path.Combine (frameworkDir, "System.Linq.Expressions.dll");
-			yield return Path.Combine (frameworkDir, "System.ComponentModel.TypeConverter.dll");
-			yield return Path.Combine (frameworkDir, "System.Console.dll");
-			yield return Path.Combine (frameworkDir, "mscorlib.dll");
-			yield return Path.Combine (frameworkDir, "System.ObjectModel.dll");
-			yield return Path.Combine (frameworkDir, "System.Runtime.Extensions.dll");
+
+			string runtimeDir = Path.GetDirectoryName (typeof (object).Assembly.Location);
+			string ncaVersion = Path.GetFileName (runtimeDir);
+			var dotnetDir = Path.GetDirectoryName (Path.GetDirectoryName (Path.GetDirectoryName (runtimeDir)));
+			var referenceDir = Path.Combine (dotnetDir, "packs", "Microsoft.NETCore.App.Ref", ncaVersion, "ref", PathUtilities.TFMDirectoryName);
+
+			yield return Path.Combine (referenceDir, "mscorlib.dll");
+			yield return Path.Combine (referenceDir, "System.Collections.dll");
+			yield return Path.Combine (referenceDir, "System.ComponentModel.TypeConverter.dll");
+			yield return Path.Combine (referenceDir, "System.Console.dll");
+			yield return Path.Combine (referenceDir, "System.Diagnostics.Tracing.dll");
+			yield return Path.Combine (referenceDir, "System.Linq.Expressions.dll");
+			yield return Path.Combine (referenceDir, "System.ObjectModel.dll");
+			yield return Path.Combine (referenceDir, "System.Reflection.Emit.dll");
+			yield return Path.Combine (referenceDir, "System.Runtime.dll");
+			yield return Path.Combine (referenceDir, "System.Runtime.Extensions.dll");
+			yield return Path.Combine (referenceDir, "System.Runtime.InteropServices.dll");
```

Comment:
Can we keep the list small? Every added assembly will slow down every test run.

Reply:
I removed a few from the common list and made them individual testcase references instead. System.Collections and System.Runtime.InteropServices are referenced by test dependencies, so I think they need to be common.